File: be-hospital/src/main/java/com/example/be_hospital/service/SpecialtyService.java
package com.example.be_hospital.service;

import com.example.be_hospital.dto.ResponseObject;
import com.example.be_hospital.dto.specialty.SpecialtyDTO;

import java.util.List;

public interface SpecialtyService {
    List<SpecialtyDTO> getAllSpecialties();
    SpecialtyDTO getSpecialtyById(int id);
    ResponseObject createSpecialty(SpecialtyDTO dto);
    ResponseObject updateSpecialty(int id, SpecialtyDTO dto);
    ResponseObject deleteSpecialty(int id);
}

File: fe-hospital/src/services/specialty.service.ts
import api from './api';

export interface SpecialtyDTO {
  id?: number;
  name: string;
  description: string;
}

export interface ResponseObject {
  status: boolean;
  message: string;
}

export const specialtyService = {
  getAllSpecialties: async (): Promise<SpecialtyDTO[]> => {
    const response = await api.get<SpecialtyDTO[]>('/specialties');
    return response.data;
  },

  getSpecialtyById: async (id: number): Promise<SpecialtyDTO> => {
    const response = await api.get<SpecialtyDTO>(`/specialties/${id}`);
    return response.data;
  },

  createSpecialty: async (dto: SpecialtyDTO): Promise<ResponseObject> => {
    const response = await api.post<ResponseObject>('/specialties', dto);
    return response.data;
  },

  updateSpecialty: async (id: number, dto: SpecialtyDTO): Promise<ResponseObject> => {
    const response = await api.put<ResponseObject>(`/specialties/${id}`, dto);
    return response.data;
  },

  deleteSpecialty: async (id: number): Promise<ResponseObject> => {
    const response = await api.delete<ResponseObject>(`/specialties/${id}`);
    return response.data;
  },
};
